languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaHttpRequestRenderer.kt

```diff
@@ -52,51 +78,184 @@ class JavaHttpRequestRenderer @Inject constructor(override val vrapTypeProvider:
+            |List<String> params = new ArrayList<>();
+            |$addingQueryParams
+            |$addingAdditionalQueryParams
+            |params.removeIf(String::isEmpty);
+            |String httpRequestPath = String.format("$stringFormat", $stringFormatArgs);
+            |if(!params.isEmpty()){
+            |   httpRequestPath += "?" + String.join("&", params);
+            |}
+        """.trimMargin().escapeAll()
+        
+        return """
+            |public HttpRequest createHttpRequest() {
+            |   HttpRequest httpRequest = new HttpRequest();
+            |   $requestPathGeneration
+            |   httpRequest.setPath(httpRequestPath); 
+            |   httpRequest.setMethod(HttpMethod.${this.method.name});
+            |   httpRequest.setHeaders(headers);
+            |   ${if(bodyName != null) "try{httpRequest.setBody(CommercetoolsJsonUtils.toJsonString($bodyName));}catch(Exception e){e.printStackTrace();}" else "" }
+            |   return httpRequest;
+            |}
+        """.trimMargin()
+    }   
+    
+    private fun Method.executeBlockingMethod() : String {
+        
+        val responseErrorsDeserialization : String = 
+                this.responses
+                        .filter { !it.statusCode.startsWith("2") }
+                        .map { responseErrorsDeserialization(it.statusCode, it.bodies[0].type.toVrapType()) }
+                        .joinToString(separator = "\n\n")
+        
+        return """
+            |public ${this.javaReturnType(vrapTypeProvider)} executeBlocking() {
+            |   HttpResponse response = CommercetoolsClient.getClient().execute(this.createHttpRequest());
+            |
+            |   $responseErrorsDeserialization
+            |   
+            |   try{
+            |       return CommercetoolsJsonUtils.fromJsonString(response.getBody(), ${this.javaReturnType(vrapTypeProvider)}.class);
+            |   }catch(Exception e){
+            |       e.printStackTrace();
+            |   }
+            |   return null;
+            |}
+        """.trimMargin().keepIndentation()
+    }
+
+    private fun responseErrorsDeserialization(statusCode : String, bodyType: VrapType) : String {
+        return """
+            |if(response.getStatusCode() == $statusCode){
+            |   try{
+            |       ${bodyType.fullClassName()} ${bodyType.simpleName().lowerCamelCase()} = CommercetoolsJsonUtils.fromJsonString(response.getBody(), ${bodyType.fullClassName()}.class);
+            |       throw new RuntimeException(${bodyType.simpleName().lowerCamelCase()}.getMessage());
+            |   }catch(Exception e){
+            |       e.printStackTrace();
+            |       throw new RuntimeException(e.getMessage());
+            |   }
+            |}
+        """.trimMargin().keepIndentation()
+    }
+    
+    private fun Method.pathArgumentsGetters() : String = this.pathArguments()
+            .map { "public String get${it.capitalize()}() {return this.$it;}" }
```

Comment:
May be we should extract a method to create the getter and setter name? We have a lot of this repetition now... WDYT?

Reply:
you mean to extract a method as in to create a Method.getter() extension method or just a simple method inside JavaHttpRequestRenderer class ?